File: frontend/src/types/staff.ts
/**
 * Staff Management Types
 */

export interface StaffInviteRequest {
    email: string;
}

export interface StaffInviteResponse {
    status: string;
    message: string;
    invitationId: string;
    recipientEmail: string;
    expiresAt: string;
}

export interface InvitationStatusResponse {
    valid: boolean;
    expired: boolean;
    recipientEmail?: string;
    inviterName?: string;
    clinicName?: string;
    message: string;
}

export interface AcceptInviteRequest {
    password: string;
    confirmPassword: string;
}

export interface AcceptInviteResponse {
    status: string;
    message: string;
    userId: string;
    accessToken: string;
    refreshToken: string;
    tokenType: string;
}

export interface StaffMember {
    id: string;
    email: string;
    firstName?: string;
    lastName?: string;
    role: string;
    isActive: boolean;
    createdAt: string;
    invitedById?: string;
}

export interface PendingInvitation {
    id: string;
    recipientEmail: string;
    createdAt: string;
    expiresAt: string;
    expired: boolean;
}

File: backend/app/schemas/staff.py
"""
Staff management schemas for invitation and onboarding.
"""

from pydantic import BaseModel, EmailStr, Field, validator
from typing import Optional
from datetime import datetime


class StaffInviteRequest(BaseModel):
    """Request schema for inviting a receptionist."""
    email: EmailStr = Field(..., description="Email address of the receptionist to invite")


class StaffInviteResponse(BaseModel):
    """Response schema after sending invitation."""
    status: str = "success"
    message: str
    invitation_id: str
    recipient_email: str
    expires_at: datetime


class AcceptInviteRequest(BaseModel):
    """Request schema for accepting an invitation."""
    password: str = Field(..., min_length=8, description="Password for the new account")
    confirm_password: str = Field(..., min_length=8, description="Password confirmation")
    
    @validator('confirm_password')
    def passwords_match(cls, v, values):
        """Validate that passwords match."""
        if 'password' in values and v != values['password']:
            raise ValueError('Passwords do not match')
        return v


class AcceptInviteResponse(BaseModel):
    """Response schema after accepting invitation."""
    status: str = "success"
    message: str
    user_id: str
    access_token: str
    refresh_token: str
    token_type: str = "bearer"


class StaffMemberResponse(BaseModel):
    """Response schema for staff member information."""
    id: str
    email: str
    first_name: Optional[str]
    last_name: Optional[str]
    role: str
    is_active: bool
    created_at: datetime
    invited_by_id: Optional[str]
    
    class Config:
        from_attributes = True


class InvitationStatusResponse(BaseModel):
    """Response schema for invitation status check."""
    valid: bool
    expired: bool
    recipient_email: Optional[str]
    inviter_name: Optional[str]
    clinic_name: Optional[str]
    message: str
